### PR TITLE
Splunk python3

### PR DESCRIPTION
**Pull request by ilaner**
**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-2262

## Description
Migration of splunk to python3

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/SplunkPy/pack_metadata.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "name": "Splunk",
     "description": "Run queries on Splunk servers.",
     "support": "xsoar",
-    "currentVersion": "2.4.16",
+    "currentVersion": "2.4.17",
```
Comment (tkatzir) [inline Packs/SplunkPy/pack_metadata.json:5]: I'd suggest bumping it a little further...
```suggestion
    "currentVersion": "2.5.0",
```

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml [hunk 2]
```diff
@@ -581,7 +581,7 @@ script:
     - contextPath: Splunk.UserMapping.SplunkUser
       description: Splunk user mapping.
       type: String
-  dockerimage: demisto/splunksdk:1.0.0.33029
+  dockerimage: devdemisto/splunksdk-py3:1.0.0.37176
```
Comment (tkatzir) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml:584]: Why is the image from `devdemisto` and not `demisto`?

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 3]
```diff
@@ -2620,8 +2582,7 @@ def main():
         splunk_parse_raw_command()
         sys.exit(0)
     service = None
-    proxy = params.get('proxy')
-    use_requests_handler = params.get('use_requests_handler')
```
Comment (tkatzir) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2624]: The `use_requests_handler` param isn't used anymore?

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml (hunk 2 above)
Comment (ilaner) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml:578]: will change once it's ready

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py (hunk 3 above)
Comment (ilaner) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2624]: If the user configures proxy, we will use only requests handler (and no longer the deprecated urllib2)

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml [hunk 4]
```diff
@@ -581,23 +575,19 @@ script:
...
   type: python
 tests:
 - SplunkPySearch_Test_default_handler
-- SplunkPySearch_Test_requests_handler
 - SplunkPy-Test-V2_default_handler
-- SplunkPy-Test-V2_requests_handler
 - Splunk-Test_default_handler
-- Splunk-Test_requests_handler
 - SplunkPy_KV_commands_default_handler
-- SplunkPy_KV_commands_requests_handler
```
Comment (yuvalbenshalom) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml:601]: what was the reason of the duplicates in the first place?

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml (hunk 2 above)
Comment (yuvalbenshalom) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml:578]: now is the time

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 5]
```diff
@@ -1887,38 +1886,6 @@ def parse_notable(notable, to_dict=False):
...
-    data = message.get('body', "") if method == 'post' else None
-    headers = dict(message.get('headers', []))
-    req = urllib2.Request(url, data, headers)  # guardrails-disable-line
-    context = ssl.create_default_context()
-
-    if VERIFY_CERTIFICATE:
-        context.verify_mode = ssl.CERT_REQUIRED
-    else:
-        context.check_hostname = False
-        context.verify_mode = ssl.CERT_NONE
```
Comment (yuvalbenshalom) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:1908]: did we test insure *and* secure?

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 6]
```diff
@@ -2,26 +2,22 @@
...
 
-urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+urllib3.disable_warnings()
 
 # Define utf8 as default encoding
-reload(sys)
-sys.setdefaultencoding('utf8')  # pylint: disable=maybe-no-member
 params = demisto.params()
 SPLUNK_TIME_FORMAT = "%Y-%m-%dT%H:%M:%S"
 VERIFY_CERTIFICATE = not bool(params.get('unsecure'))
```
Comment (yuvalbenshalom) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:23]: how us this being handled in the new version?

File: Packs/SplunkPy/TestPlaybooks/playbook-SplunkPy-Test-V2.yml [hunk 7]
```diff
@@ -85,7 +85,7 @@ tasks:
         simple: demisto
       batch_limit: {}
       earliest_time:
-        simple: -1000d
+        simple: -1500d
```
Comment (yuvalbenshalom) [inline Packs/SplunkPy/TestPlaybooks/playbook-SplunkPy-Test-V2.yml:88]: what was this change needed?

File: Packs/SplunkPy/ReleaseNotes/2_5_0.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### SplunkPy
```
Comment (yuvalbenshalom) [inline Packs/SplunkPy/ReleaseNotes/2_5_0.md:3]: let's decide about this together.

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
see my comments.
also, please connect the jira issue that is related and add deprecation for splunk pre release.(and not it in the PR explanation)

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py (hunk 6 above)
Comment (ilaner) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:23]: fixed. we will use the requests handler if user does not want to verify certificate

File: Packs/SplunkPy/ReleaseNotes/2_5_0.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+- Migrated from python 2.7 to python 3.10.
+- Upgraded the splunk-sdk to 1.7.2.
+- Removed the `use_requests_handler` parameter. If selected to use `proxy` or `insecure` parameters, the integration will use the requests handler.
+- Updated the Docker image to: *demisto/splunksdk-py3:1.0.0.37176*.
```
Comment (ilappe) [inline Packs/SplunkPy/ReleaseNotes/2_5_0.md:7]: please update according the decision to use the handler always 

also the name of this file and the json file should be 3_0_0

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 10]
```diff
@@ -373,7 +375,7 @@ def fetch_notables(service, mapper, cache_object=None, enrich_notables=False):
                     'query: {} is: {}.'.format(last_run_time, now, fetch_query, len(incidents))
     extensive_log(debug_message)
 
-    if not enrich_notables:
+    if not enrich_notables or not cache_object:
```
Comment (ilappe) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:378]: what will be here if this are the first fetch and the enrichment is enabled ?
cache_object will not be None?

**Review by ilappe — CHANGES_REQUESTED** [review]

File: Packs/SplunkPy/ReleaseNotes/2_5_0.md (hunk 9 above)
Comment (ilaner) [reply Packs/SplunkPy/ReleaseNotes/2_5_0.md]: changed

**Review by ilappe — APPROVED** [review]
GRATE WORK!!!
